packages/agreement-process/src/services/agreementService.ts

```diff
@@ -757,7 +765,16 @@ export function agreementServiceBuilder(
         `Retrieving consumer document ${documentId} from agreement ${agreementId}`
       );
       const agreement = await retrieveAgreement(agreementId, readModelService);
-      assertRequesterIsConsumerOrProducer(agreement.data, authData);
+      const delegation = await retrieveDelegationByDelegateId(
+        authData.organizationId,
+        readModelService
+      );
```

Comment:
We can skip this read-model read in case the requester is a producer or consumer. I would move this inside the assertion and retrieve the delegation only if it is needed

Reply:
[344e131](https://github.com/pagopa/interop-be-monorepo/pull/1157/commits/344e131da53e09dfd7989614ab131ac162c95373)